lib/Db/Wopi.php

```diff
@@ -96,12 +96,10 @@ public function getWopiForToken($token) {
 			return false;
 		}
 
-		//TODO: validate.
-		if ($row['expiry'] > \time()) {
+		if ($row['expiry'] <= \time()) {
 			// Expired token!
-			//http_response_code(404);
-			//$wopi->deleteBy('id', $row['id']);
-			//return false;
+			\http_response_code(404);
```

Comment:
I don't think this side effect is needed. It's enough for `getWopiForToken` to returns `false` to controller.
And the controller should set HTTP status on false. As I see it's already done everywhere except https://github.com/owncloud/richdocuments/blob/master/lib/Controller/DocumentController.php#L879

Reply:
@karakayasemi seems @VicDeo is right, you can remove it here and a line above, in that call just returning `false` should be sufficient as upper layer is handling response.

Reply:
Yes, I agree with you guys. But, the inconsistency of the method that bothers me here. The upper if block is using http_response_code also. I would remove both of them then. But, I can not be sure any possible side effect of this action.

Reply:
As @vicdeo stated, only one usage of the method has not any result check. I added a check for this exception place to return proper htp response code.

Reply:
Do not worry, it is handled outside of that class